docs/internals/protocol/typedesc.rst

```diff
@@ -73,7 +73,8 @@ Object Shape Descriptor
         //   1 << 0: the field is implicit
         //   1 << 1: the field is a link property
         //   1 << 2: the field is a link
-        uint8           flags;
+        //   7 << 3: Cardinality
```

Comment:
Perhaps it'd be cleaner to split the upper and lower cardinality into two binary flags instead?

Reply:
@1st1 ^

Reply:
changed